### ISSUE TITLE
ID Token is missing after expiration

### ISSUE THREAD

**Issue by aso1datov** [AzureAD/microsoft-authentication-library-for-js#5863]
### Core Library

MSAL.js v2 (@azure/msal-browser)

### Core Library Version

2.35.0

### Wrapper Library

MSAL React (@azure/msal-react)

### Wrapper Library Version

1.5.5

### Public or Confidential Client?

Public

### Description

After the token is refreshed, the ID token disappears from Authentication Result.

Response before refresh (initial) 

```json
{
   ...
    "account": {
        "homeAccountId": "370fe65a-1daa-4dad-aa7d-f2df0fbae9b9.2b4de1bd-251e-4878-bdb8-5180f7d15525",
        "environment": "login.windows.net",
        "tenantId": "2b4de1bd-251e-4878-bdb8-5180f7d15525",
        "username": "a.soldatov@***.com",
        "localAccountId": "370fe65a-1daa-4dad-aa7d-f2df0fbae9b9",
        "name": "Aleksandr Soldatov",
        "idTokenClaims": {
            "aud": "a5a8765a-474f-4a75-bbac-ec3141984541",
            "iss": "https://login.microsoftonline.com/2b4de1bd-251e-4878-bdb8-5180f7d15525/v2.0",
            "iat": 1680864363,
            "nbf": 1680864363,
            "exp": 1680868263,
            "name": "Aleksandr Soldatov",
            "nonce": "084cc69d-2784-43f2-a753-927869216706",
            "oid": "370fe65a-1daa-4dad-aa7d-f2df0fbae9b9",
            "preferred_username": "a.soldatov@convergint.com",
            "rh": "0.AVoAveFNKx4leEi9uFGA99FVJVp2qKVPR3VKu6zsMUGYRUFaAKU.",
            "sub": "xSBqz-c8thH4AF93wR8hTe3iq1fDNc-uKtg-u4WdGV8",
            "tid": "2b4de1bd-251e-4878-bdb8-5180f7d15525",
            "uti": "jY9_R6mFJ0GCXbIgQ9RZAQ",
            "ver": "2.0"
        }
    },
    "idToken": "Omitted to reduce the amount of code",
    "idTokenClaims": {
        "aud": "a5a8765a-474f-4a75-bbac-ec3141984541",
        "iss": "https://login.microsoftonline.com/2b4de1bd-251e-4878-bdb8-5180f7d15525/v2.0",
        "iat": 1680864363,
        "nbf": 1680864363,
        "exp": 1680868263,
        "name": "Aleksandr Soldatov",
        "nonce": "084cc69d-2784-43f2-a753-927869216706",
        "oid": "370fe65a-1daa-4dad-aa7d-f2df0fbae9b9",
        "preferred_username": "a.soldatov@***.com",
        "rh": "0.AVoAveFNKx4leEi9uFGA99FVJVp2qKVPR3VKu6zsMUGYRUFaAKU.",
        "sub": "xSBqz-c8thH4AF93wR8hTe3iq1fDNc-uKtg-u4WdGV8",
        "tid": "2b4de1bd-251e-4878-bdb8-5180f7d15525",
        "uti": "jY9_R6mFJ0GCXbIgQ9RZAQ",
        "ver": "2.0"
    },
    "accessToken": "Omitted to reduce the amount of code",
    "fromCache": true,
    "expiresOn": "2023-04-07T11:53:09.000Z",
    "correlationId": "9385b3dd-f6a4-4a3f-9759-16e350fba6fa",
    "requestId": "",
    "extExpiresOn": "2023-04-07T12:55:15.000Z",
    "familyId": "",
    "tokenType": "Bearer",
    "state": "",
    "cloudGraphHostName": "",
    "msGraphHost": "",
    "fromNativeBroker": false
}

```

Response after refresh:

```json
{
    ...
    "account": {
        "homeAccountId": "370fe65a-1daa-4dad-aa7d-f2df0fbae9b9.2b4de1bd-251e-4878-bdb8-5180f7d15525",
        "environment": "login.windows.net",
        "tenantId": "2b4de1bd-251e-4878-bdb8-5180f7d15525",
        "username": "a.soldatov@***.com",
        "localAccountId": "370fe65a-1daa-4dad-aa7d-f2df0fbae9b9",
        "name": "Aleksandr Soldatov",
        "idTokenClaims": {
            "aud": "a5a8765a-474f-4a75-bbac-ec3141984541",
            "iss": "https://login.microsoftonline.com/2b4de1bd-251e-4878-bdb8-5180f7d15525/v2.0",
            "iat": 1680862657,
            "nbf": 1680862657,
            "exp": 1680866557,
            "name": "Aleksandr Soldatov",
            "nonce": "555be7e9-5f35-4372-ab72-014c008cccb9",
            "oid": "370fe65a-1daa-4dad-aa7d-f2df0fbae9b9",
            "preferred_username": "a.soldatov@convergint.com",
            "rh": "0.AVoAveFNKx4leEi9uFGA99FVJVp2qKVPR3VKu6zsMUGYRUFaAKU.",
            "sub": "xSBqz-c8thH4AF93wR8hTe3iq1fDNc-uKtg-u4WdGV8",
            "tid": "2b4de1bd-251e-4878-bdb8-5180f7d15525",
            "uti": "lhMNy9Js9E60y0VmRIKFAQ",
            "ver": "2.0"
        }
    },
    "idToken": "",
    "idTokenClaims": {},
    "accessToken": "Omitted to reduce the amount of code",
    "fromCache": true,
    "expiresOn": "2023-04-07T11:23:30.000Z",
    ...
}
```

### MSAL Configuration

```javascript
{
    "auth": {
        "clientId": "a5a8765a-474f-4a75-bbac-ec3141984541",
        "redirectUri": "/",
        "authority": "https://login.microsoftonline.com/2b4de1bd-251e-4878-bdb8-5180f7d15525",
        "navigateToLoginRequestUrl": false,
        "postLogoutRedirectUri": "http://localhost:3000"
    },
    "cache": {
        "cacheLocation": "localStorage",
        "storeAuthStateInCookie": false
    }
}
```

### Relevant Code Snippets

```javascript
async function getAuthTokens() {
  const msalInstance = getMsalInstance();
  const account = msalInstance.getActiveAccount();

  if (account !== null) {
    const idTokenClaims = account.idTokenClaims as { exp: number };
    const forceRefresh = new Date(idTokenClaims.exp * 1000) < new Date();
    const accessTokenRequest: SilentRequest = {
      account,
      scopes: loginRequest.scopes,
    };

    try {
      const { accessToken, idToken } = await msalInstance.acquireTokenSilent({
        ...accessTokenRequest,
        forceRefresh,
      });

      return { accessToken, idToken };
    } catch (error) {
      if (error instanceof InteractionRequiredAuthError) {
        await msalInstance.acquireTokenRedirect(accessTokenRequest);
      } else {
        throw error;
      }
    }
  }
}
```

### Reproduction Steps

1. Login to the application
2. Wait until the ID token expires
3. Check response from acquire token silent. The ID token would be empty

### Expected Behavior

Response from `acquireTokenSilent` contains ID token

### Identity Provider

Azure AD / MSA

### Browsers Affected (Select all that apply)

Chrome, Firefox, Edge, Safari

### Regression

@azure/msal-browser 2.34.0

### Source

Internal (Microsoft)

**Comment by jnnck** [comment]
I like to confirm we're experiencing the same issue. We were able to "solve" it by putting v2.34.0 as a fixed dependency

**Comment by tnorling** [comment]
Thanks for raising this, I was able to confirm this is a bug in the latest version. I'll link a PR once I have one ready.

**Comment by tnorling** [comment]
Fix was merged and will be included in our next release currently scheduled for May 1st. Thanks again for bringing this to our attention!

**Comment by belcurv** [comment]
Appears this is fixed w/ 2.36.0, which is published and can be installed today. This bug hit my team pretty hard - lots of hours spent trying to reproduce. Would have benefitted from seeing 2.36.0 in the official releases. Today, 2.35.0 is the latest msal-browser version listed.
